lib/database/migrations/20240424092020-add-qc-flag-effective-period-columns.js

```diff
@@ -0,0 +1,262 @@
+        }
+        groupedQcFlagsForDataPass[data_pass_id][run_number][dpl_detector_id].push(flag);
+    }
+
+    for (const perRunDetector of Object.values(groupedQcFlagsForDataPass)) {
+        for (const perDetector of Object.values(perRunDetector)) {
+            for (const qcFlags of Object.values(perDetector)) {
+                await _reconsturctEffectivePeriodForFlagsInTheSameScope(queryInterface, qcFlags, { transaction });
+            }
+        }
+    }
+
+    const [qcFlagsPerSimulationPass] = await queryInterface.sequelize.query(`
+        SELECT
+            qcf.id,
+            \`from\`,
+            \`to\`,
+            run_number,
+            dpl_detector_id,
+            sp.id as simulation_pass_id
+        FROM quality_control_flags as qcf
+        INNER JOIN simulation_pass_quality_control_flag as spqcf
+            ON qcf.id = spqcf.quality_control_flag_id
+        INNER JOIN simulation_passes as sp
+            ON sp.id = spqcf.simulation_pass_id
+        ORDER by qcf.created_at ASC
+        ;
+    `, { transaction }) || [];
+
+    const groupedQcFlagsForSimulationPass = {};
+    for (const flag of qcFlagsPerSimulationPass) {
+        const {
+            simulation_pass_id,
+            run_number,
+            dpl_detector_id,
+        } = flag;
+        if (!groupedQcFlagsForSimulationPass[simulation_pass_id]) {
+            groupedQcFlagsForSimulationPass[simulation_pass_id] = {};
+        }
+        if (!groupedQcFlagsForSimulationPass[simulation_pass_id][run_number]) {
+            groupedQcFlagsForSimulationPass[simulation_pass_id][run_number] = {};
+        }
+        if (!groupedQcFlagsForSimulationPass[simulation_pass_id][run_number][dpl_detector_id]) {
+            groupedQcFlagsForSimulationPass[simulation_pass_id][run_number][dpl_detector_id] = [];
+        }
+        groupedQcFlagsForSimulationPass[simulation_pass_id][run_number][dpl_detector_id].push(flag);
+    }
+
+    for (const perRunDetector of Object.values(groupedQcFlagsForSimulationPass)) {
+        for (const perDetector of Object.values(perRunDetector)) {
+            for (const qcFlags of Object.values(perDetector)) {
+                await _reconsturctEffectivePeriodForFlagsInTheSameScope(queryInterface, qcFlags, { transaction });
+            }
+        }
+    }
+};
+
+/**
+ * Reset effective period of all QC flags
+ * @param {QueryInterface} queryInterface sequelize interface
+ * @return {Promise<void>} promise
+ */
+const _resetQcFlagEffectivePeriod = async (queryInterface, { transaction }) => {
+    await queryInterface.sequelize.query('DELETE FROM quality_control_flag_effective_periods;', { transaction });
+    await queryInterface.sequelize.query(
+        `
+        INSERT INTO quality_control_flag_effective_periods(flag_id, \`from\`, \`to\`)
+            SELECT qcf.id, qcf.\`from\`, qcf.\`to\` from quality_control_flags qcf;`,
+        { transaction },
+    );
+};
```

Comment:
Why reset? You just created the table, it's empty. You can directly name it populate
Also, once again, please make effort in your documentation to actually explain what you are doing!!! `Reset effective period of all QC flags` is completely unclear. `Create effective period for every existing QC flag over its whole period` for example

Reply:
applied